docs/design/Chem Integration.md

```diff
@@ -0,0 +1,69 @@
+# Loose binary identification with Upgrade Assistant
+
+It is not uncommon in large code bases to have a collection of dependencies that have been persisted as only the `.dll` files with no known provenance. These files are considered "loose binaries". This makes it challenging to identify what a binary is while upgrading where an updated version is often needed to support .NET Standard/Core/5. In order to solve this, this document proposes integration with a data source that can help match these loose assemblies to known NuGet packages.
```

Comment:
There's an unstated assumption here that the reader is familiar with our upgrade guidance and understands that they should update their dependencies as part of the upgrade process. Where possible, we actually prefer this is done before they run upgrade assistant to smooth out the process.

Maybe something simple in the intro like:
`One of the earliest steps in the process of upgrading to .NET Standard/Core/5 is to upgrade your dependencies.`